docs/integrations/databases/supabase.mdx

```diff
@@ -3,170 +3,666 @@ title: Integrate Supabase with Clerk
+      const supabase = createClient(
+        process.env.NEXT_PUBLIC_SUPABASE_URL ?? "",
+        process.env.NEXT_PUBLIC_SUPABASE_KEY ?? "",
+        {
+          global: {
+            headers: {
+              Authorization: `Bearer ${supabaseAccessToken}`,
+            },
+          },
+        }
+      );
+      return supabase;
+    };
+
+    export default function Home() {
+      const { getToken, userId } = useAuth();
+
+      const getSupabaseClient = async () => {
+        // Replace with your JWT template name
+        const supabaseAccessToken = await getToken({ template: "supabase" });
+        if (!supabaseAccessToken)
+          throw new Error("Failed to get token from JWT template");
+        return await supabaseClient(supabaseAccessToken);
+      };
+
+      const [posts, setPosts] = useState<any>(null);
+      const listPosts = async () => {
+        const client = await getSupabaseClient();
+        // Fetches all posts scoped to the user
+        // Replace "Posts" with your table name
+        const { data, error } = await client.from("Posts").select();
+        setPosts(error || data);
+      };
+
+      const inputRef = useRef<HTMLInputElement>(null);
+      const sendPost = async () => {
+        if (!inputRef.current?.value) return;
+        const client = await getSupabaseClient();
+        const { data, error } = await client.from("Posts").insert({
+          user_id: userId,
+          // Replace content with whatever field you want
+          content: inputRef.current?.value,
+        });
+      };
+
+      return (
+        <>
+          <div style={{ display: "flex", flexDirection: "column" }}>
+            <input style={{ color: "black" }} type="text" ref={inputRef} />
+            <button onClick={sendPost}>Send Post</button>
+            <button onClick={listPosts}>Fetch Posts</button>
+          </div>
+          <h2>Posts</h2>
+          {!posts ? (
+            <p>No posts</p>
+          ) : (
+            <ul>
+              {posts.map((post: any) => (
+                <li key={post.id}>{post.content}</li>
+              ))}
+            </ul>
+          )}
+        </>
+      );
+    }
+    ```
+    </CodeBlockTabs>
+1. Try out your application. When you visit the page with your component, you'll be required to sign in. Try creating and fetching content.
+
+</Steps>
 
-```json
-CREATE POLICY "Authenticated users can update their own todos" 
-	ON public.todos FOR UPDATE USING (
-		auth.role() = 'authenticated'::text
-	) WITH CHECK (
-		requesting_user_id() = user_id
-	);
-```
 
 ## Next steps
 
-- Get started with our official [clerk-supabase-starter](https://github.com/clerk/clerk-supabase-starter) repo
-- Check out our [Next.js + Supabase + Clerk tutorial](https://clerk.com/blog/nextjs-supabase-todos-with-multifactor-authentication)
+- Get started with Clerk's official [clerk-supabase-starter](https://github.com/clerk/clerk-supabase-starter) repo
```

Comment:
I'm assuming before this goes live we'll want to wait for an updated starter to match the integration path & an updated blog with the same?

Reply:
I think we can remove these links for now, then add a link to the updated starter when it's done as a small followup PR. Thoughts?

Reply:
Depends on how we want to view this as done. 

I'll spin up the starter now and see if there's anything that's needed to be added there for a quick update to be able to keep it. 

As far as the blog, we can unlink the previous one, and re-add once a new one is created.